xclim/indices/_threshold.py

```diff
@@ -336,14 +336,13 @@ def cold_spell_total_length(
     thresh = convert_units_to(thresh, tas)
 
     cond = compare(tas, op, thresh, constrain=("<", "<="))
-    max_l = rl.resample_and_rl(
+    out = rl.resample_and_rl(
         cond,
         resample_before_rl,
         rl.windowed_run_count,
-        window=1,
+        window=window,
         freq=freq,
     )
-    out = max_l.where(max_l >= window, 0)
```

Comment:
Before, we had: 

"Count all time steps in spells of any given length, then it that total length is not at least `window`, set this to 0. " 

Now we simply have: "Total number of time steps that are present in spells with at least `window` time steps."